### PR TITLE
feat: add delete icon to delete a task instead of just clicking on the task tile

### LINKED ISSUE #100: Add a delete icon next to the task
**Is your feature request related to a problem? Please describe.**
Delete option appears when the user clicks on the task. There is no clear indicator like an icon for the same. 

**Describe the solution you'd like**
A delete icon next to the task should be added. Delete task dialog box should appear when the icon is clicked instead of the task.

**Additional context**
Current Version

### PR DESCRIPTION
**Pull request by bharat-1809**
### Description
Add a delete button on the task tile in the relation page

Fixes #100 

### Flutter Channel:
- [ ] I have used the Flutter Beta channel on my local machine 

### Type of Change:

- Code

**Code/Quality Assurance Only**

- New feature (non-breaking change which adds functionality pre-approved by mentors)

### How Has This Been Tested?
On a physical device
Steps:
- Go to the Tasks tab in the relations page by clicking on the relation page icon in the bottom navigation bar
- Here's a demo of the same

### Checklist:

- [x] My PR follows the style guidelines of this project
- [x] I have performed a self-review of my own code or materials
- [x] I have commented my code or provided relevant documentation, particularly in hard-to-understand areas

**Code/Quality Assurance Only**
- [x] My changes generate no new warnings

### CODE REVIEW

File: lib/screens/home/pages/relation/relation_page.dart [hunk 1]
```diff
@@ -140,43 +140,50 @@ class _RelationPageState extends State<RelationPage> {
...
                       ),
-                    ),
-                    Text(task.description),
-                  ],
-                ),
+                      Text(task.description),
+                    ],
+                  ),
+                  IconButton(
+                    icon: Icon(Icons.delete, color: Colors.grey[700],),
```
Comment (bartekpacia) [inline lib/screens/home/pages/relation/relation_page.dart]: please use `dartfmt` to format the code. If you used `dartfmt` this should be on a new line

File: lib/screens/home/pages/relation/relation_page.dart [hunk 2]
```diff
@@ -140,43 +140,50 @@ class _RelationPageState extends State<RelationPage> {
...
+                    onPressed: () {
+                      showDialog(
+                        context: context,
+                        builder: (context) => AlertDialog(
+                          title: Text("Delete task"),
+                          content: Text("Are you sure you want to delete the task?"),
+                          actions: [
```
Comment (bartekpacia) [inline lib/screens/home/pages/relation/relation_page.dart:219]: add second button (on the left) saying `Cancel`

**Review by bartekpacia — DISMISSED** [review]

**Comment by bartekpacia** [comment]
This is great! Thanks:)
Take a look at my 2 comments, fix that stuff and it should be ready to be merged

**Comment by bharat-1809** [comment]
@bartekpacia sure :)

File: lib/screens/home/pages/relation/relation_page.dart (hunk 1 above)
Comment (bharat-1809) [reply lib/screens/home/pages/relation/relation_page.dart]: Done

File: lib/screens/home/pages/relation/relation_page.dart (hunk 2 above)
Comment (bharat-1809) [reply lib/screens/home/pages/relation/relation_page.dart:219]: Added a cancel button to the delete dialog box
@bartekpacia please have a look
Thanks :)
Screenshot_20200624-150739

**Review by HaripriyaB — DISMISSED** [review]
nice work @bharat-1809 🎉

File: lib/screens/home/pages/relation/relation_page.dart (hunk 2 above)
Comment (techno-disaster) [reply lib/screens/home/pages/relation/relation_page.dart:219]: looks like you didn't push these changes?

Comment (bharat-1809) [reply lib/screens/home/pages/relation/relation_page.dart:219]: > looks like you didn't push these changes?

Ohh sorry :sweat_smile:

Comment (bharat-1809) [reply lib/screens/home/pages/relation/relation_page.dart:219]: @Techno-Disaster  Done

**Review by HaripriyaB — DISMISSED** [review]

**Comment by Akanksha1212** [comment]
Looks good. Great work @bharat-1809

**Review by robotjellyzone — DISMISSED** [review]
LGTM  🎉

**Comment by HaripriyaB** [comment]
2 reviews are done so moving this to needs testing.

**Comment by HaripriyaB** [comment]
Hey @bharat-1809 I found a bug pls check the below gif:
Record-2020-06-30-21-40-44-92746
When I click the bin icon and delete it keeps loading and nothing actually happens. Can anyone confirm whether this issue valid?? @anitab-org/qa-team

File: lib/screens/home/pages/relation/relation_page.dart [hunk 3]
```diff
@@ -140,43 +140,59 @@ class _RelationPageState extends State<RelationPage> {
...
+                            FlatButton(
+                              onPressed: () {
+                                Navigator.of(context).pop();
+                              },
+                              child: Text("Cancel"),
+                            ),
+                            FlatButton(
+                              child: Text("Delete"),
+                              onPressed: () {
+                                // bloc.add(TaskDeleted(state.relation, task.id));
```
Comment (HaripriyaB) [inline lib/screens/home/pages/relation/relation_page.dart]: Not sure why you have commented this...May be this is the cause that the delete task is not executed and the progress bar spins forever.

Comment (techno-disaster) [reply lib/screens/home/pages/relation/relation_page.dart]: yep. This should be it

Comment (techno-disaster) [reply lib/screens/home/pages/relation/relation_page.dart]: @bharat-1809 Can you check this part?

**Review by techno-disaster — CHANGES_REQUESTED** [review]

File: lib/screens/home/pages/relation/relation_page.dart (hunk 3 above)
Comment (bharat-1809) [reply lib/screens/home/pages/relation/relation_page.dart]: Oh sorry :sweat_smile:

Comment (bharat-1809) [reply lib/screens/home/pages/relation/relation_page.dart]: Its done

**Review by HaripriyaB — DISMISSED** [review]
I think perfect now 🎉

**Comment by HaripriyaB** [comment]
The changes made in this PR were tested locally. Following are the results:

1. Code review - Done 

2. All possible responses (positive and negative tests) were tested as below:

  * Tested deleting a to do task and completed task
    _Screenshot/gif_:  
Record-2020-07-01-01-49-14-92746

    _Expected Result_:  Delete operation to work on both the cases
    _Actual Result_: As expected

3. Status of PR Changed to: Ready to Merge ✔️

**Review by robotjellyzone — DISMISSED** [review]

**Review by techno-disaster — CHANGES_REQUESTED** [review]
@bharat-1809 can you fix this merge conflict?

**Comment by bharat-1809** [comment]
@Techno-Disaster Its done. please have a look :)

**Review by isabelcosta — APPROVED** [review]
Good work @bharat-1809 👏 
Thank you for your contribution!

**Review by bartekpacia — APPROVED** [review]
looks good, thanks:)

**Review by techno-disaster — APPROVED** [review]
LGTM

**Pull request merged**